From: Modestas Vainius <modax@debian.org>
Subject: Fix mysql_config call for Debian MySQL
Last-Update: 2012-05-21

* As of 5.5.23, mysql_config_pic is shipped inside libmysqld-pic package.
* As of 5.5.23, -L/usr/lib/mysql is no longer in --libmysqld-libs so linker
  can't find -lmysqld_pic anymore.

-- Found MySQL: Usage: /usr/bin/mysql_config [OPTIONS]
Options:
        --cflags         [-I/usr/include/mysql  -DBIG_JOINS=1  -fno-strict-aliasing   -DUNIV_LINUX -DUNIV_LINUX]
        --include        [-I/usr/include/mysql]
        --libs           [-rdynamic -L/usr/lib/mysql -lmysqlclient]
        --libs_r         [-rdynamic -L/usr/lib/mysql -lmysqlclient_r]
        --plugindir      [/usr/lib/mysql/plugin]
        --socket         [/var/run/mysqld/mysqld.sock]
        --port           [0]
        --version        [5.1.58]
        --libmysqld-libs [-rdynamic -L/usr/lib/mysql -lmysqld -ldl -lwrap -lrt], -rdynamic -L/usr/lib/mysql -lmysqlclient
-- Found MySQL Embedded: Usage: /usr/bin/mysql_config [OPTIONS]
Options:
        --cflags         [-I/usr/include/mysql  -DBIG_JOINS=1  -fno-strict-aliasing   -DUNIV_LINUX -DUNIV_LINUX]
        --include        [-I/usr/include/mysql]
        --libs           [-rdynamic -L/usr/lib/mysql -lmysqlclient]
        --libs_r         [-rdynamic -L/usr/lib/mysql -lmysqlclient_r]
        --plugindir      [/usr/lib/mysql/plugin]
        --socket         [/var/run/mysqld/mysqld.sock]
        --port           [0]
        --version        [5.1.58]
        --libmysqld-libs [-rdynamic -L/usr/lib/mysql -lmysqld -ldl -lwrap -lrt], -rdynamic -L/usr/lib/mysql -lmysqld_pic -ldl -lwrap -lrt -lpthread

ii  libmysqlclient-dev            5.1.58-1                      MySQL database development files
--- a/cmake/modules/FindMySQLAmarok.cmake
+++ b/cmake/modules/FindMySQLAmarok.cmake
@@ -13,11 +13,11 @@
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
 if(NOT WIN32)
-    find_program(MYSQLCONFIG_EXECUTABLE NAMES mysql_config mysql_config5 PATHS ${BIN_INSTALL_DIR} ~/usr/bin /usr/local/bin)
+    find_program(MYSQLCONFIG_EXECUTABLE NAMES mysql_config_pic mysql_config mysql_config5 PATHS ${BIN_INSTALL_DIR} ~/usr/bin /usr/local/bin)
 endif(NOT WIN32)
 
 if(MYSQLCONFIG_EXECUTABLE)
-    exec_program(${MYSQLCONFIG_EXECUTABLE} ARGS --variable=pkgincludedir RETURN_VALUE _return_VALUE OUTPUT_VARIABLE MYSQL_INCLUDE_DIR)
+    exec_program(${MYSQLCONFIG_EXECUTABLE} ARGS --include RETURN_VALUE _return_VALUE OUTPUT_VARIABLE MYSQL_INCLUDE_DIR)
     exec_program(${MYSQLCONFIG_EXECUTABLE} ARGS --cflags RETURN_VALUE _return_VALUE OUTPUT_VARIABLE MYSQL_CFLAGS)
     exec_program(${MYSQLCONFIG_EXECUTABLE} ARGS --libs RETURN_VALUE _return_VALUE OUTPUT_VARIABLE MYSQL_LIBRARIES)
     exec_program(${MYSQLCONFIG_EXECUTABLE} ARGS --libmysqld-libs RETURN_VALUE _return_VALUE OUTPUT_VARIABLE MYSQL_EMBEDDED_LIBSTEMP)
@@ -36,6 +36,9 @@ if(MYSQLCONFIG_EXECUTABLE)
 
     if(MYSQLD_PIC_SEPARATE)
         string(REPLACE "lmysqld" "lmysqld_pic" MYSQL_EMBEDDED_LIBRARIES ${MYSQL_EMBEDDED_LIBSTEMP})
+        # Hack for Debian. mysql_config{,_pic} as of 5.5.23 does not refer to
+        # -L/usr/lib/mysql where mysqld_pic actually lives (see #673574)
+        set(MYSQL_EMBEDDED_LIBRARIES "-L/usr/lib/mysql ${MYSQL_EMBEDDED_LIBRARIES}")
     else(MYSQLD_PIC_SEPARATE)
         set(MYSQL_EMBEDDED_LIBRARIES ${MYSQL_EMBEDDED_LIBSTEMP})
     endif(MYSQLD_PIC_SEPARATE)
